src/app/catalog/product/components/product-detail/product-detail.component.ts

```diff
@@ -3,34 +3,77 @@
  * SPDX-License-Identifier: MIT
  */
 
-import { ChangeDetectionStrategy, Component, OnInit, signal } from '@angular/core';
+import {
+  ChangeDetectionStrategy,
+  ChangeDetectorRef,
+  Component,
+  inject,
+  OnDestroy,
+  OnInit,
+  signal,
+} from '@angular/core';
 import { Store } from '@ngrx/store';
 
 import { selectCurrentProductDetails } from '@app/catalog/product/store/product.selectors';
 import { navigate } from '@app/shared/navigation/navigation.actions';
 import { StateWithCatalog } from '@app/catalog/store/catalog.reducer';
 import { loadProductDetails } from '@app/catalog/product/store/product.actions';
 import { ActivatedRoute } from '@angular/router';
+import { ShoppingCartStore } from '@app/shared/signal-store/shopping-cart.store';
+import { Product } from '@models/product';
+import { Subscription } from 'rxjs';
 
 @Component({
   selector: 'app-product-detail',
   changeDetection: ChangeDetectionStrategy.OnPush,
   templateUrl: './product-detail.component.html',
   styleUrls: ['./product-detail.component.scss'],
 })
-export class ProductDetailComponent implements OnInit {
+export class ProductDetailComponent implements OnInit, OnDestroy {
   product$ = this.store.select(selectCurrentProductDetails);
+  private loadedProduct: Product;
+  private loadProductSubscription: Subscription;
+
+  public shoppingCartSignalStore = inject(ShoppingCartStore);
 
   productNumber = signal(1);
 
-  constructor(private store: Store<StateWithCatalog>, private route: ActivatedRoute) {}
+  constructor(
+    private store: Store<StateWithCatalog>,
+    private route: ActivatedRoute,
+    private cdr: ChangeDetectorRef,
+  ) {}
 
   backToProductOverview(): void {
     this.store.dispatch(navigate({ url: '/' }));
   }
 
   showConfirmation(): void {
-    this.store.dispatch(navigate({ url: '/order' }));
+    // it's quite strange to navigate through a store's effect,but this is an old pattern in the codebase
```

Comment:
Why is it strange?

Regardless, I would prefer to remove the comment as this is code that we show to a public audience.

Reply:
removed